core/src/main/java/cz/xtf/core/openshift/OpenShift.java

```diff
@@ -982,7 +993,12 @@ public PodShell podShell(Pod pod) {
 	 */
 	public Waiter clean() {
 		for (CustomResourceDefinitionContextProvider crdContextProvider : OpenShift.getCRDContextProviders()) {
-			customResource(crdContextProvider.getContext()).delete(getNamespace());
+			CustomResourceDefinitionContext crdContext = crdContextProvider.getContext();
+			if (getCRDItems(crdContext, getNamespace()).size() > 0) {
+				log.debug("DELETE :: " + crdContext.getName() + " instances");
+				customResource(crdContext).delete(getNamespace());
+				new SimpleWaiter(() -> getCRDItems(crdContext, getNamespace()).size() == 0, TimeUnit.SECONDS, 20, "Cleaning CRDs.").waitFor();
```

Comment:
This might not be enough (20 seconds) as Operator can create a lot of objects and pod which took time to terminate. I'm testing this atm.

Reply:
Ok, 20s doesn't seem to be a problem here, my problem was a finalizer...

Reply:
No waiting should be executed within the function. But you are right that there is a bug and `isProjectClean()` doesn't take custom resources into account. `isProjectClean()` should be fixed for this purpose. For the finalizer, is there possibility to do the same as with regular resources (cc @dsimansk )?